panel-app/src/components/store.ts

```diff
@@ -185,23 +173,35 @@ function formatState() {
 
 	const formattedMetadata =
 		metadata &&
-		Object.keys(metadata).map((key) => {
+		Object.keys(metadata).reduce((result, key) => {
```

Comment:
I see this logic with `reduce` a few times in this file. Can we have a function `getFormattedMetadata()`?
They are all the same, no?
3/4